packages/contracts/src/dollar/libraries/LibDirectGovernanceFarmer.sol

```diff
@@ -340,34 +342,37 @@ contract DirectGovernanceFarmer is ReentrancyGuard {
             stakingShareId
         );
 
-        // STEP 1 : Withdraw Ubiquity Staking Shares to get back uAD3CRV-f LPs
-        //address staking = ubiquityManager.stakingContractAddress();
+        // STEP 1: Withdraw Ubiquity Staking Shares to get back uAD3CRV-f LPs
+        // address staking = ubiquityManager.stakingContractAddress();
         IStakingShare(stakingShare).setApprovalForAll(staking, true);
         IStaking(staking).removeLiquidity(stake.lpAmount, stakingShareId);
         IStakingShare(stakingShare).setApprovalForAll(staking, false);
 
-        uint256 lpTokenAmount = IERC20(ubiquity3PoolLP).balanceOf(
+        uint256 lpTokenAmount = IERC20(data.ubiquity3PoolLP).balanceOf(
             address(this)
         );
-        uint256 governanceTokenAmount = IERC20(manager.governanceTokenAddress())
-            .balanceOf(address(this));
+        uint256 governanceTokenAmount = IERC20(
+            data.manager.governanceTokenAddress()
+        ).balanceOf(address(this));
 
-        // STEP2 : Withdraw  3Crv LPs from meta pool to get back Ubiquity Dollar, DAI, USDC or USDT
+        // STEP2: Withdraw  3Crv LPs from meta pool to get back Ubiquity Dollar, DAI, USDC or USDT
 
-        IERC20(ubiquity3PoolLP).approve(
-            depositZapUbiquityDollar,
+        IERC20(data.ubiquity3PoolLP).approve(
+            data.depositZapUbiquityDollar,
             lpTokenAmount
         );
-        tokenAmounts = IDepositZap(depositZapUbiquityDollar).remove_liquidity(
-            ubiquity3PoolLP,
-            lpTokenAmount,
-            [uint256(0), uint256(0), uint256(0), uint256(0)]
-        ); //[Ubiquity Dollar, DAI, USDC, USDT]
-        IERC20(ubiquityDollar).safeTransfer(msg.sender, tokenAmounts[0]);
-        IERC20(token0).safeTransfer(msg.sender, tokenAmounts[1]);
-        IERC20(token1).safeTransfer(msg.sender, tokenAmounts[2]);
-        IERC20(token2).safeTransfer(msg.sender, tokenAmounts[3]);
-        IERC20(manager.governanceTokenAddress()).safeTransfer(
+        tokenAmounts = IDepositZap(data.depositZapUbiquityDollar)
+            .remove_liquidity(
+                data.ubiquity3PoolLP,
+                lpTokenAmount,
+                [uint256(0), uint256(0), uint256(0), uint256(0)]
+            ); //[Ubiquity Dollar, DAI, USDC, USDT]
+
+        IERC20(data.ubiquityDollar).safeTransfer(msg.sender, tokenAmounts[0]);
+        IERC20(data.token0).safeTransfer(msg.sender, tokenAmounts[1]);
+        IERC20(data.token1).safeTransfer(msg.sender, tokenAmounts[2]);
+        IERC20(data.token2).safeTransfer(msg.sender, tokenAmounts[3]);
```

Comment:
It is good this is covered by tests, took me a while to go up and down and compare indexes, especially that indexes differ, e.g. token0 is in tokenAmounts[1]. Perhaps if token0, token1, token2 were renamed to token_DAI, token_USDC, token_USDT it would be more readable and less error prone. But I understand you move to facet so the original names were not changed.

Reply:
absolutely, one of the purposes is that im keeping on original as possible with minimal changes, planning to do extra changes, but later

Reply:
ok got it